compiler/src/iree/compiler/Codegen/Common/MaterializeEncodingPass.cpp

```diff
@@ -240,12 +254,23 @@ static FailureOr<MaterializeEncodingInfo> chooseEncodingInfo(
   return encodingInfo;
 }
 
+static FailureOr<MaterializeEncodingValueInfo>
+chooseDynamicEncodingInfoVMVXMicrokernels(RankedTensorType tensorType,
+                                          OpBuilder &builder, Location loc) {
+  // For now just create Values equal to 1.
+  // TODO: create a new vmvx.get_tile_sizes op here.
+  MaterializeEncodingValueInfo info;
+  info.innerTileSizes = SmallVector<Value>(
+      tensorType.getRank(), builder.create<arith::ConstantIndexOp>(loc, 1));
+  return info;
+}
```

Comment:
My understanding is:

1. We know the tiling sizes when materializing encodings. Because we have all the information at this level.
2. The codegen compiler should be able to get the information. We need the information when doing tile+distribution. Because we needs to infer the packed types. So far, the inner tiling sizes are all static, so we can carry the information in an ArrayRef<int64_t> array, and pass it to the patterns.
3. Microkernel does not know inner tiling sizes, we need a way to pass the information to microkernels.

This solution looks fine to me, because it can be a placeholder for filling gaps. But does it really address your dynamic tiling sizes issue? They are constants ops, and 

I'm thinking if we can use `util.global` ops when materializing the encoding.. Then all the inner tile sizes can be queried by util.global.load ops. We can do a broader discussion after people are back from vacation.

In this way, we're able to provide needed information for codegen. I.e., DispatchWorkgroupCountFromSetEncodingOp pattern can look into the variables to infer packed layout. And the microkernels can also query the inner tile size from the `util.global` ops.

@benvanik should weigh in because he has more understanding about `util.global` ops.

Reply:
(as discussed, landing this PR and the subsequent ones fleshing out the tile-sizes query, will help make this conversation concrete)